src/bundle/Resources/public/admin/js/ibexa-automated-translation.js

```diff
@@ -1,57 +1,61 @@
+    const error = container.querySelector('.ibexa-automated-translation-error');
+    const sourceLabel = container.querySelector('.ibexa-field-edit--ezboolean .ibexa-data-source__label');
+
+    const toggleClass = (event) => {
+        const target = event.target;
+        const input = target.querySelector('input[type="checkbox"]');
+
+        if (input.checked) {
+            input.removeAttribute('checked');
+            target.classList.remove('is-checked');
         } else {
-            $(this).addClass('is-checked');
-            $input.attr('checked', 'checked');
+            target.classList.add('is-checked');
+            input.setAttribute('checked', true);
         }
-        return false;
-    });
+    };
 
-    $container.find(".ibexa-data-source__label.is-checked").each(() => {
-        $form.find('.ibexa-btn--create-translation').prop("disabled", false);
-    });
-
-    $container.find(".ibexa-dropdown__source select option:selected").each(() => {
-        $form.find('.ibexa-btn--create-translation').prop("disabled", false);
-    });
-
-    $("form[name=add-translation]").submit(function () {
-        let targetLang = $("select[name=add-translation\\[language\\]]").val();
-        let sourceLang = $("select[name=add-translation\\[base_language\\]]").val();
-        let mapping = $container.data('languages-mapping');
-        let $serviceSelector = $("#add-translation_translatorAlias");
-        let serviceAlias = $serviceSelector.val();
-        if ($serviceSelector.is("[type=checkbox]") && !$serviceSelector.is(":checked")) {
-            serviceAlias = '';
-        }
+    const submitForm = () => {
+        const targetLang = form.querySelector('select[name="add-translation[base_language]"]').value;
+        const sourceLang = form.querySelector('select[name="add-translation[language]"]').value;
+        const { languagesMapping } = container.dataset;
+        const serviceSelector = form.querySelector('#add-translation_translatorAlias');
+        const serviceAlias = serviceSelector.value;
 
-        if (!serviceAlias.length) {
+        if ((serviceSelector.type === 'checkbox' && !serviceSelector.checked) || !serviceAlias.length) {
             return true;
         }
 
-        let translationAvailable = (typeof sourceLang === 'undefined' || -1 !== $.inArray(sourceLang, mapping[serviceAlias])) && (-1 !== $.inArray(targetLang, mapping[serviceAlias]));
-        if (false === translationAvailable) {
-            $error.removeClass("invisible");
-            if ($container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").hasClass('is-checked')) {
-                $container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").click();
+        const translationAvailable =
+            (typeof sourceLang === 'undefined' || sourceLang.indexOf(languagesMapping[serviceAlias]) !== -1) &&
+            targetLang.indexOf(targetLang, languagesMapping[serviceAlias]) !== -1;
```

Comment:
```suggestion
        const serviceLang = languagesMapping[serviceAlias];
        const translationAvailable = (typeof sourceLang === 'undefined' || sourceLang.includes(serviceLang)) && targetLang.includes(serviceLang);
```
or sth like this, not sure of the name `serviceLang`, but based on serviceAlias and languagesMapping names I guess it keeps some kind of language